packages/react/src/components/Comments/ThreadsSidebar.tsx

```diff
@@ -0,0 +1,188 @@
+  }
+
+  return threads;
+}
+
+/**
+ * The ThreadsSidebar component can be used to display a list of comments in a sidebar.
+ *
+ * This component is similar to Google Docs "Show All Comments" sidebar (cmd+option+shift+A)
+ */
+export function ThreadsSidebar(props: {
+  filter?: "open" | "resolved";
+  // TODO: Should be implemented for both floating and sidebar views
+  // maxCommentsPerThread?: number;
+  sort?: "position" | "newest" | "oldest";
+}) {
+  const editor = useBlockNoteEditor();
+
+  if (!editor.comments) {
+    throw new Error("Comments plugin not found");
+  }
+
+  const state = useUIPluginState(
+    editor.comments.onUpdate.bind(editor.comments)
+  );
+  const selectedThreadId = state?.selectedThreadId;
+
+  const threads = useThreads(editor);
+
+  const threadPositions = useMemo<
+    | Map<
+        string,
+        {
+          from: number;
+          to: number;
+        }
+      >
+    | undefined
+  >(
+    () =>
+      editor.comments?.plugin.getState(editor.prosemirrorState)
+        ?.threadPositions,
+    [editor.comments?.plugin, editor.prosemirrorState]
+  );
+
+  useEffect(() => {
+    if (!selectedThreadId) {
+      return;
+    }
+
+    const selectedThreadPosition = threadPositions?.get(selectedThreadId);
+    if (!selectedThreadPosition) {
+      return;
+    }
+
+    (
+      editor.prosemirrorView?.domAtPos(selectedThreadPosition.from).node as
+        | Element
+        | undefined
+    )?.scrollIntoView({
+      behavior: "smooth",
+      block: "center",
+    });
+
+    const selectedThread = threads.get(selectedThreadId);
+    if (!selectedThread) {
+      return;
+    }
+
+    if (selectedThread.resolved) {
+      editor.prosemirrorView?.dispatch(
+        editor.prosemirrorState.tr.setSelection(
+          TextSelection.create(
+            editor.prosemirrorState.doc,
+            selectedThreadPosition.from,
+            selectedThreadPosition.to
+          )
+        )
+      );
+      editor.setForceSelectionVisible(true);
+    }
+  }, [
+    editor,
+    editor.comments.plugin,
+    editor.prosemirrorState,
+    editor.prosemirrorState.doc,
+    editor.prosemirrorView,
+    selectedThreadId,
+    threadPositions,
+    threads,
+  ]);
+
+  const threadElements = useMemo(() => {
+    const threadsArray = Array.from(threads.values());
+
+    const sortedThreads = sortThreads(editor, threadsArray, props.sort);
+
+    const openThreads: ThreadData[] = [];
+    const resolvedThreads: ThreadData[] = [];
+
+    for (const thread of sortedThreads) {
+      if (!thread.resolved) {
+        openThreads.push(thread);
+      } else {
+        resolvedThreads.push(thread);
+      }
+    }
+
+    const threadDataToElement = (thread: ThreadData) => {
+      const threadPosition = threadPositions?.get(thread.id);
+
+      return (
+        <Thread
+          key={thread.id}
+          thread={thread}
+          view={"sidebar"}
+          selected={thread.id === selectedThreadId}
+          highlightedText={
```

Comment:
wow this is super fancy :) didn't think we'd get this in

- we might need to truncate it? for long parts of text
- in google docs I think they store the original content (like it was when the first comment was added). I think there are definitely some good arguments for that, but maybe ok to keep out of scope for now